src/options-component/DataTab.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui-next/styles';
+import classNames from 'classnames';
+import { FormControl } from 'material-ui-next/Form';
+import Select from 'material-ui-next/Select';
+import TextField from 'material-ui-next/TextField';
+import { InputLabel } from 'material-ui-next/Input';
+import { MenuItem } from 'material-ui-next/Menu';
+import DataTabCheckBoxes from './DataTabCheckBoxes';
+import strings from './utils';
+
+const styles = {
+    formControl: {
+        minWidth: 200,
+    },
+    inputLabeltextSize: {
+        fontSize: 13,
+    },
+    textFields: {
+        marginRight: 20,
+        width: 100,
+    },
+    formControlCoverWholeRow: {
+        marginRight: '100%',
+    },
+    textFieldCoverWholeRow: {
+        marginRight: '55%',
+        width: 200,
+    },
+    divBorder: {
+        borderBottom: '1px solid rgb(158, 158, 158)',
+        paddingBottom: 35,
+    },
+};
+
+const renderTextFields = (classes, tabContent, onChange) => {
+    return tabContent.map(([key, value], i) => (
+        <TextField
+            className={
+                i % 2 === 0
+                    ? classes.textFields
+                    : classes.textFieldCoverWholeRow
+            } // 2nd and 4th TextField component pads out rest of the row space
+            InputLabelProps={{ className: classes.inputLabeltextSize }}
+            label={strings.data[key]}
+            onChange={event => {
+                onChange(key, event.target.value);
+            }} // TODO (?) use parseInt on number? onChange will always pass a string
+            type={i % 2 === 0 ? 'number' : 'string'} // 1st and 3rd TextField component are number specific
+            value={value}
+            key={i}
+        />
+    ));
+};
+
+const renderSelectFields = (classes, tabContent, onChange) => {
+    //render first two select components
+    // call			{renderTextFields(classes, textFields, onChange)} (?)
+    //render last select component
+};
+
+const DataTab = props => {
+    const { classes, tabContent, onChange } = props;
+    let textFields = Object.entries(tabContent).slice(5, 9); // slice the prop object, to render DataTab's textfield values
+    return (
+        <div className={classes.divBorder}>
+            <DataTabCheckBoxes tabContent={tabContent} onChange={onChange} />
+
+            <FormControl className={classes.formControl}>
+                <InputLabel className={classes.inputLabeltextSize}>
+                    {' '}

```

Comment:
avoid using empty spaces for padding

Reply:
I'm thinking that each tab should be smart enough and know which props its going to display.  So an internal data-properties object that contains all the props that will be displayed instead of the list as you have in index.js. In fact, I think maybe we should make use of react component state, for the moment, although that will most likely get replaced with redux.